### PR TITLE
Remove BaseIndex class.

### PR DESCRIPTION
**Pull request by s2quake**
https://github.com/planetarium/libplanet/issues/3645

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ To be released.
  -  Removed the '#nullable disable' from the Libplanet.Store project. [[#3644]]
  -  Removed the '#nullable disable' from the Libplanet.RocksDBStore project.
     [[#3651]]
+ -  Removed `BaseIndex` class and changed `BlockSet` base class
+    from `BaseIndex` to `IDictionary`.  [[#3686]]
```
Comment (greymistcube) [inline CHANGES.md:19]:
```suggestion
    from `BaseIndex<BlockHash, Block>` to `IDictionary<BlockHash, Block>`.  [[#3686]]
```

**Comment by greymistcube** [comment]
Eh, I think implementing `IDictionary<K, V>` is still an overkill since `BlockSet` is supposed to be readonly really.
How about changing it to `IReadOnlydictionary<K, V>`?

**Comment by s2quake** [comment]
> Eh, I think implementing `IDictionary<K, V>` is still an overkill since `BlockSet` is supposed to be readonly really. How about changing it to `IReadOnlydictionary<K, V>`?

The `_blocks` field in `BlockChain` has a write operation code, so I changed its type to `BlockSet`.
see [this](https://github.com/planetarium/libplanet/pull/3686/files#diff-ecd6275ea8686534dc160d341079910ede8da3d2ec020236582bdb0906cddc21R59)

**Comment by greymistcube** [comment]
I guess I missed it somehow. Very well, I still think there are larger design issues pertaining to this problem, such as its scope and features, but I think this is enough for now.

**Review by greymistcube — APPROVED** [review]

**Pull request merged**